snap/snapcraft.yaml

```diff
@@ -36,6 +36,37 @@ parts:
     override-build: |
       mkdir -p $CRAFT_PART_INSTALL/bin
 
+      # Install NodeJS and ZAP tool for arm builds
+      if [[ $SNAP_ARCH == "arm64" ]]; then
+        set -x
```

Comment:
It is best to move this code and the relevant build packages to a separate part for better caching and clarity.
I think a quicker way to test that is to run it for arm64.

Reply:
in my case, testing it on amd64 is much quicker, do you mean the same?

Reply:
Yes.

Reply:
After moving this code and the relevant build packages to a separate part, the amd64 works as expected, but arm64 build has a new error:
```
npm ci
:: /usr/bin/node: 1: ELF: not found                                                                                                                                                                 
:: /usr/bin/node: 2: Syntax error: ")" unexpected                                                                                                                                                         
'override-build' in part 'nodejs-zap' failed with code 2.
```

Reply:
In case it wasn't clear: what I meant by testing it quicker was to test the code against amd64. That is by replacing
`if [[ $SNAP_ARCH == "arm64" ]]; then`
with
`if [[ $SNAP_ARCH == "amd64" ]]; then`
and then building it on amd64.

Reply:
yes, that's what I have done

Reply:
This error might caused by incompatibility between nodejs and npm, and here is a potential solution: https://stackoverflow.com/a/72256347

Reply:
The ELF error is because the code is installing an x86 nodejd inside the arm64 snap. It is not related to moving the installation to a separate part, but that change may have surfaced the underlying issue hidden because of the cached installation. Note that when something is installed outside the part (`/opt/node-v12.22.12-linux-x64/`), a cleaning of the part or even a full clean from inside the LXD instance will not remove those files. A clean from outside the instance will be effective because it removed the LXD instance completely. To avoid that caching issue, build resources should be installed beneath the path's build directory.